### PR TITLE
Place command

### PR DESCRIPTION
**Pull request by ricardo6**
added tags table, tag_places table and places table.
search by place name, search by tags

### CODE REVIEW

File: src/main/java/org/jala/efeeder/places/CreatePlaceCommand.java [hunk 1]
```diff
@@ -1,28 +1,36 @@
...
 /**
  *
  * @author ricardo_ramirez
  */
 @Command
 public class CreatePlaceCommand implements CommandUnit {
 	private static final String UPDATE_PLACE_QUERY = "INSERT INTO places(name, description, phone, direction, image_link) values(?, ?, ?, ?, ?)";
-
+	private static final String UPDATE_TAG_QUERY = "INSERT INTO tags(name) values(?)";
+    private static final String GET_TAG_ID_QUERY = "SELECT id FROM tags WHERE name=?";
```
Comment (DanitzaMachicado) [inline src/main/java/org/jala/efeeder/places/CreatePlaceCommand.java]: Fix tabs

File: src/main/java/org/jala/efeeder/places/CreatePlaceCommand.java [hunk 2]
```diff
@@ -31,11 +39,54 @@ public Out execute(In parameters) throws Exception {
...
 		placeId = generatedKeys.getInt(1);
-
+		String arrTags = parameters.getParameter("tags");
+		/* tag list for this place */
+		String[] tags = arrTags.split(":");
+		
+		/* tag name will be one-off */
+		for (String tag : tags) {
+			int tagId = getTagId(tag);
+			if(tagId > 0)
```
Comment (DanitzaMachicado) [inline src/main/java/org/jala/efeeder/places/CreatePlaceCommand.java]: According SCiQuest's code conventions "always use {} for statements if or else"

File: src/main/java/org/jala/efeeder/places/CreatePlaceCommand.java [hunk 3]
```diff
@@ -31,11 +39,54 @@ public Out execute(In parameters) throws Exception {
...
 								parameters.getParameter("address"), parameters.getParameter("image_link"));
 		return OutBuilder.response("application/json", JsonConverter.objectToJSON(place));
+		
+	}
+	private int getTagId(String tag) throws Exception {
+		int tagId = -1; //not exits tag with this id
+		PreparedStatement preparedStatement = connection.prepareStatement(GET_TAG_ID_QUERY);
+		preparedStatement.setString(1, tag);
+		ResultSet rSet = preparedStatement.executeQuery();
+		if (rSet.next())
```
Comment (DanitzaMachicado) [inline src/main/java/org/jala/efeeder/places/CreatePlaceCommand.java]: Please use {}

File: src/main/java/org/jala/efeeder/places/CreatePlaceCommand.java [hunk 4]
```diff
@@ -31,11 +39,54 @@ public Out execute(In parameters) throws Exception {
...
+		PreparedStatement prepareStatement = connection.prepareStatement(UPDATE_TAG_QUERY, RETURN_GENERATED_KEYS);
+		prepareStatement.setString(1, tag);
+		prepareStatement.executeUpdate();
+		ResultSet generatedKeys = prepareStatement.getGeneratedKeys();
+		generatedKeys.next();
+		return generatedKeys.getInt(1);
+	}
+
+	private void updateTagPlaces(int tagId, int placeId) throws SQLException {
+		PreparedStatement prepareStatement1 = connection.prepareStatement(UPDATE_TAG_PLACES_QUERY);
```
Comment (DanitzaMachicado) [inline src/main/java/org/jala/efeeder/places/CreatePlaceCommand.java]: Improvement name

File: src/main/java/org/jala/efeeder/places/SearchPlaceCommand.java [hunk 5]
```diff
@@ -50,11 +52,10 @@ public Out execute(In parameters) throws Exception {
...
-	private String getWhere(String term) {
-		if (term == null || term.isEmpty()) {
+	private String prepareQuery(String term, int ENTRIES_BY_PAGE, int page) {
+		if (term == null || term.isEmpty()) 
 			return "";
-		}
-		term += "%";
-		return String.format("WHERE name like \"%s\" ", term);
+		else 
+			return String.format(SEARCH_PLACES_QUERY_PATHERM, term, term, ENTRIES_BY_PAGE, page);
```
Comment (DanitzaMachicado) [inline src/main/java/org/jala/efeeder/places/SearchPlaceCommand.java]: Please use {}

**Review by DanitzaMachicado — CHANGES_REQUESTED** [review]

File: src/main/java/org/jala/efeeder/suggestion/SuggestionsCommand.java [hunk 6]
```diff
@@ -17,7 +17,7 @@
  */
 @Command
 public class SuggestionsCommand implements CommandUnit {
-	private static final String TOP_FIVE_PLACES_QUERY = "SELECT * FROM places ORDER BY created_at DESC limit 5";
+	private static final String TOP_FIVE_PLACES_QUERY = "SELECT * FROM places ORDER BY created_at DESC limit 10";
```
Comment (Ramir0) [inline src/main/java/org/jala/efeeder/suggestion/SuggestionsCommand.java:24]: top five places has 10 places as limit ?

**Review by 0x3 — APPROVED** [review]

**Review by Ramir0 — APPROVED** [review]

**Review by DanitzaMachicado — APPROVED** [review]

**Pull request merged**